farmerbot/parser/validator.go

```diff
@@ -0,0 +1,97 @@
+package parser
+
+import (
+	"fmt"
+	"slices"
+
+	substrate "github.com/threefoldtech/tfchain/clients/tfchain-client-go"
+	"github.com/threefoldtech/tfgrid-sdk-go/farmerbot/internal"
+)
+
+// wrapper for validateInput
+func ValidateConfig(input internal.Config, network string) error {
+	manager := substrate.NewManager(internal.SubstrateURLs[network]...)
+	subConn, err := manager.Substrate()
+	if err != nil {
+		return err
+	}
+	defer subConn.Close()
+	return validateInput(input, subConn)
+}
+
+// validateInput validates that included, excluded and priority nodes are in the farm
+func validateInput(input internal.Config, sub internal.Substrate) error {
+	nodes, err := sub.GetNodes(input.FarmID)
+	if err != nil {
+		return fmt.Errorf("couldn't retrieve node for %d : %v", input.FarmID, err)
+	}
+	nodesMap := make(map[uint32]bool)
+	for _, node := range nodes {
+		nodesMap[node] = true
+	}
+	if len(input.IncludedNodes) != 0 {
+		if err := validateWithIncludedNodes(input, nodesMap); err != nil {
```

Comment:
i think what thabet meant was to create a validator function for every set of nodes eg (validateIncludedNodes, validateExcludedNodes,...)